File: jre/src/main/java/com/azure/data/cosmos/serialization/hybridrow/layouts/LayoutResolverSimple.java
//------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All rights reserved.
//------------------------------------------------------------

package com.azure.data.cosmos.serialization.hybridrow.layouts;

import com.azure.data.cosmos.serialization.hybridrow.SchemaId;

public final class LayoutResolverSimple extends LayoutResolver {
    private tangible.Func1Param<SchemaId, Layout> resolver;

    public LayoutResolverSimple(tangible.Func1Param<SchemaId, Layout> resolver) {
        this.resolver = (SchemaId arg) -> resolver.invoke(arg);
    }

    @Override
    public Layout Resolve(SchemaId schemaId) {
        return this.resolver.invoke(schemaId.clone());
    }
}

File: src/Serialization/HybridRow/Layouts/LayoutResolverSimple.cs
// ------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All rights reserved.
// ------------------------------------------------------------

namespace Microsoft.Azure.Cosmos.Serialization.HybridRow.Layouts
{
    using System;

    public sealed class LayoutResolverSimple : LayoutResolver
    {
        private readonly Func<SchemaId, Layout> resolver;

        public LayoutResolverSimple(Func<SchemaId, Layout> resolver)
        {
            this.resolver = resolver;
        }

        public override Layout Resolve(SchemaId schemaId)
        {
            return this.resolver(schemaId);
        }
    }
}
